plug.vim

```diff
@@ -747,6 +744,10 @@ function! s:update_impl(pull, force, args) abort
     echohl None
   endif
 
+  let python = (has('python') || has('python3'))
+      \ && (!s:nvim || (has('nvim') && has('vim_starting'))) && !s:is_win && !has('win32unix')
+  let ruby = has('ruby') && !has('nvim') && (v:version >= 703 || v:version == 702 && has('patch374'))
```

Comment:
How about we check `has('nvim')` only once when defining `s:nvim` and see `s:nvim` instead in other places? I think we can reorganize the expressions as follows to make them slightly less confusing

```vim
  let python = (has('python') || has('python3')) && !s:is_win && !has('win32unix')
      \ && (!s:nvim || has('vim_starting')) 
  let ruby = has('ruby') && (v:version >= 703 || v:version == 702 && has('patch374'))
      \ && !s:nvim
```

- has python & not windows & (neovim installer n/a or vim is starting)
    - We can just strip out vim_starting part when jobwait is ready
- has ruby & vim version not too old & neovim installer n/a